### PR TITLE
Updated gen2 docs and updated import paths

### PR DESCRIPTION
**Pull request by renebrandel**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by elorzafe — APPROVED** [review]
Looks good auth part

**Review by abdallahshaban557 — DISMISSED** [review]
LGTM

File: src/directory/directory.mjs [hunk 1]
```diff
@@ -2127,23 +2128,24 @@ export const directory = {
...
                 },
+                // Uncommented because this feature won't be ready for re:invent
+                // {
+                //   path: 'src/pages/gen2/build-a-backend/data/offline/index.mdx'
+                // },
                 {
-                  path: 'src/pages/gen2/build-a-backend/data/offline/index.mdx'
-                }
+                  path: 'src/pages/gen2/build-a-backend/data/migration/index.mdx'
+                },
```
Comment (heatheramz) [inline src/directory/directory.mjs:2148]:
```suggestion
                }
```

File: src/pages/gen2/build-a-backend/data/connect-to-API/index.mdx [hunk 2]
```diff
@@ -13,206 +13,295 @@ export function getStaticProps(context) {
...
+In this guide, you will connect your application code to the backend API using the Amplify Libraries. We will review how to install and configure Amplify Libraries.
 
-You can use the Amplify library to connect GraphQL APIs created in various ways:
-- Amplify vNext-generated API (backend/data.ts)
-- Amplify Classic-generated API (`backend/api/<api-name\>/schema.graphql`)
-- any existing AWS AppSync GraphQL API
-- GraphQL APIs hosted using API Gateway
+Before you begin, you will need:
+- your cloud sandbox with an Amplify Data resource up and running (`npx amplify sandbox`)
+- A frontend application set up. For instructions on setting up an app, see [Project Setup: Create your application](https://docs.amplify.aws/lib/project-setup/create-application/q/platform/js/)
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/connect-to-API/index.mdx:20]: @renebrandel general comment that links on this page should be updated to new pages - would project set up here go to a Gen2 get started?

File: src/pages/gen2/build-a-backend/data/connect-to-API/index.mdx [hunk 3]
```diff
@@ -13,206 +13,295 @@ export function getStaticProps(context) {
...
+</BlockSwitcher>
+
+## Conclusion
+
+Congratulations! You have finished the **Connect your app code to the API** guide. In this guide, you reviewed the required configuration and installed and configured Amplify Libraries. You also reviewed some quick ways to customize the data client with different authorization modes and custom headers.
+
+## Next steps
+
+Our recommended next steps include using the API to mutate and query data. You can also review how to subscribe to real-time events to look for mutations in your data. Some resources that will help with this work include:
+- [Create, update, and delete application data](/lib/graphqlapi/mutate-data/q/platform/js/)
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/connect-to-API/index.mdx:305]: These should be updated to Gen2 links

File: src/pages/gen2/build-a-backend/data/mutate-data/index.mdx [hunk 4]
```diff
@@ -13,407 +13,148 @@ export function getStaticProps(context) {
...
 
-# Read and write real-time data
+In this guide, you will learn how to create, update, and delete your data using Amplify Libraries' Data client.
 
-## Configure the API client code
+Before you begin, you will need:
 
-### Use with Amplify Data
-You can get a client using the defined schema and get type-safe CRUDL operations to talk to your backend database in your Node.js app.
+- An [application connected to the API](/lib/graphqlapi/connect/q/platform/js/)
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/mutate-data/index.mdx:20]: Update to Gen2 link

File: src/pages/gen2/build-a-backend/data/mutate-data/index.mdx [hunk 5]
```diff
@@ -13,407 +13,148 @@ export function getStaticProps(context) {
...
+
+## Conclusion
+
+Congratulations! You have finished the **Create, update, and delete application data** guide. In this guide, you created, updated, and deleted your app data through the GraphQL API.
+
+## Next steps
+
+Our recommended next steps include using the API to query data and subscribe to real-time events to look for mutations in your data. Some resources that will help with this work include:
+
+- [Read application data](/lib/graphqlapi/query-data/q/platform/js/)
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/mutate-data/index.mdx:159]: Again flag for updating links to Gen 2

File: src/pages/gen2/build-a-backend/data/query-data/index.mdx [hunk 6]
```diff
@@ -13,10 +13,269 @@ export function getStaticProps(context) {
...
-## List data
-### Apply filters in list queries
-### Paginate over list queries
-## Get a specific item
-## Call a custom GraphQL query
+You can read application data using the Amplify Data client. In this guide, we will review the difference between reading data and getting data, how to filter query results to get just the data you need, and how to paginate results to make your data more manageable. We will also show you how to cancel these requests when needed.
+
+Before you begin, you will need:
+
+- An [application connected to the API](/lib/graphqlapi/connect/q/platform/js/)
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/query-data/index.mdx:21]: link update

File: src/pages/gen2/build-a-backend/data/query-data/index.mdx [hunk 7]
```diff
@@ -13,10 +13,269 @@ export function getStaticProps(context) {
...
+```
+
+## Conclusion
+
+Congratulations! You have finished the **Read application data** guide. In this guide, you learned how to read your data through `get` and `list` queries.
+
+## Next steps
+
+Our recommended next steps include subscribing to real-time events to look for mutations in your data and continuing to build out and customize your information architecture for your data. Some resources that will help with this work include:
+
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/query-data/index.mdx:277]: link update

File: src/pages/gen2/build-a-backend/data/subscribe-data/index.mdx [hunk 8]
```diff
@@ -13,38 +13,221 @@ export function getStaticProps(context) {
...
 
-# Customize real-time subscriptions
+In this guide, we will outline the benefits of enabling real-time data integrations and how to set up and filter these subscriptions. We will also cover how to unsubscribe from subscriptions.
 
-### Change subscription access-level
-By default, real-time updates are on for all `a.model()`, which means customers receive real-time updates if their authorization rules allows it. You can also turn off subscriptions for that model or make the real-time updates public, receivable by all subscribers.
+Before you begin, you will need:
+
+
+- An [application connected to the API](/lib/graphqlapi/connect/q/platform/js/)
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/subscribe-data/index.mdx:21]: link update

File: src/pages/gen2/build-a-backend/data/subscribe-data/index.mdx [hunk 9]
```diff
@@ -13,38 +13,221 @@ export function getStaticProps(context) {
...
 ```
+
+## Conclusion
+
+Congratulations! You have finished the **Subscribe to real-time events** guide. In this guide, you set up subscriptions for real-time events and learned how to filter and cancel these subscriptions when needed.
+
+## Next steps
+
+Our recommended next steps include continuing to build out and customize your information architecture for your data. Some resources that will help with this work include:
+
```
Comment (heatheramz) [inline src/pages/gen2/build-a-backend/data/subscribe-data/index.mdx:230]: link update

**Review by heatheramz — CHANGES_REQUESTED** [review]
@renebrandel just a heads up that links can be updated and one other nit for the the directory

File: src/pages/[platform]/build-a-backend/graphqlapi/modify-amplify-generated-resources/index.mdx [hunk 10]
```diff
@@ -1,19 +1,10 @@
...
-                'angular',
-                'flutter',
-                'javascript',
-                'nextjs',
-                'react',
-                'react-native',
-                'swift',
-                'vue'
-              ]
+  title: 'Set up Amplify GraphQL API',
```
Comment (heatheramz) [inline src/pages/[platform]/build-a-backend/graphqlapi/modify-amplify-generated-resources/index.mdx:4]: @renebrandel The above platforms would need to be listed if you want those visiting react etc to see the JS content

**Review by heatheramz — APPROVED** [review]
Per our slack convo we'll follow up on links later

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Pull request merged**